Icons

jrvollmer opened the pull request.

jkira1 committed:
created icons for the faults with descriptions

jrvollmer committed:
Merge branch 'sw-122/IOBoards' into icons

jrvollmer committed:
Implemented faults

jrvollmer committed:
Merge branch 'main' into icons

jrvollmer committed:
Updated Box heights so that icons stay in place

jrvollmer committed:
Merge branch 'main' into icons

jrvollmer committed:
Moved images and added solar car connection

jrvollmer committed:
Added solar car connection here too

jrvollmer committed:
Reverted package and package-lock

jrvollmer committed:
Merge branch 'main' into icons

jrvollmer committed:
Removed unused fault

jrvollmer committed:
Removed unused images

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
+import WirelessCommsLostImage from "./Images/Wireless Comms Lost.png";
+import HighTemperatureImage from "./Images/High Temperature.png";
 
 export default function Faults(props) {
```
I found most of the functions in this component pretty confusing, can we write some method headers and/or inline comments?

jrvollmer committed:
One last unused image

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
+            eStopStrings.push("External");
+        }
+
+        if(eStopStrings.length > 0) {
```
The bottom part looks pretty identical to a few other methods, perhaps we can split this off into it's own private function. (iirc private functions in javascript technically don't exist but convention is to preceed them with _ i.e 
```javascript
const myPublicFunction = () => {};
const _myPrivateFunction = () => {};
```

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
+    }
+
+    const checkTemps = () => {
+        return (props.data?.motor_temp[0] > 100) || (props.data?.driverIO_temp[0] > 100) ||
```
Can we set the magic numbers as a constant stored somewhere? Don't wanna go find all the instances of sensor bounds whenever they needa change

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
+        let highTempStrings = [];
+        let lowTempStrings = [];
+
+        if(props.data?.motor_temp[0] > 100) {
```
Same comment about magic numbers

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
+        let highTempString = "";
+        let lowTempString = "";
+
+        if(highTempStrings.length > 0) {
```
Is this if statement necessary?

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
+        let highTempString = "";
+        let lowTempString = "";
+
+        if(highTempStrings.length > 0) {
+            if(highTempStrings.length == 1) {
+                highTempString = highTempStrings[0] + " temp is high";
+            } else if(highTempStrings.length == 2) {
+                highTempString = highTempStrings[0] + " and " + highTempStrings[1] + " temps are high";
+            } else {
+                highTempString = highTempStrings[0];
+                for(let i=1; i < highTempStrings.length - 1; i++) {
+                    highTempString += ", " + highTempStrings[i];
+                }
+                highTempString += ", and " + highTempStrings[highTempStrings.length - 1] + " temps are high";
+            }
+        }
```
```suggestion
          if(highTempStrings.length == 1) {
              highTempString = highTempStrings[0] + " temp is high";
          } else if(highTempStrings.length == 2) {
              highTempString = highTempStrings[0] + " and " + highTempStrings[1] + " temps are high";
          } else {
              highTempString = highTempStrings[0];
              for(let i=1; i < highTempStrings.length - 1; i++) {
                    highTempString += ", " + highTempStrings[i];
              }
              highTempString += ", and " + highTempStrings[highTempStrings.length - 1] + " temps are high";
          }
```
Formatting is probably bad but you get the point. If `length == 1` then `length > 0` so no need for 2 ifs

EUdds reviewed (changes requested):
Good work - a few notes. Feel free to message me with questions.

jrvollmer committed:
Added data format constants and script to auto-populate it

jrvollmer committed:
Updated comments and README

jrvollmer committed:
Nothing special

jrvollmer committed:
assert type json

jrvollmer committed:
Fixed start script

jrvollmer committed:
Re-open socket after lost connection and track lost connections

If the client errors out, it will attempt to reopen the socket at
one second intervals. Also, solar_car_connection is now an array,
so lost connections between the solar car and engineering dashboard
can be tracked in the graphs. Right now, the most recent value of
solar_car_connection is set to false whenever connection is lost,
and true is pushed to the front of the array whenever new data is
unpacked. Because of this, only the element of solar_car_connection
that corresponds to the last packet received before connection was
lost will be false.

jrvollmer committed:
Added update-data-constants script

jrvollmer committed:
Removed some TODOs

jrvollmer committed:
Made function for forming Strings containing non-nominal values

jrvollmer committed:
Moved some repetitive code into another private function

jrvollmer committed:
Added method headers

jrvollmer committed:
My favorite parts: comments and debugging

jrvollmer committed:
I suck

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
-import { Heading, HStack, Text, VStack } from "@chakra-ui/react";
+import { Box, Tooltip, Image, SimpleGrid } from "@chakra-ui/react";
```
We could write an autocoder for this to find `*.png` in the Images folder and create the `import X from "./Images/X.png` automatically in it's own component. Not needed for this PR but just a thought

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
+            } else {
+                return eStopStrings[0] + " E-stop was pressed";
+            }
+        }
```
```suggestion
        if(eStopStrings.length > 0) {
            buf = "eStopStrings[0] ";
            for(let i=i; i < eStopStrings.length; i++) { // Starting at the 2nd index because first index isn't prepped with a comma
                buf += `, and ${eStopStrings[i].toLowerCase()} `;
               }
               buf += "E-stop was pressed";
               return buf;
           }
```

That way if you add more estop strings you don't have to rewrite this function

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
+            failsafeStrings.push("Relay");
+        }
+
+        if(failsafeStrings.length > 0) {
```
Consider doing a for loop like shown above

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,334 @@
+        if(props.data?.motor_temp[0] > 100) {
+            highTempStrings.push("Motor");
+        }
+        if(props.data?.driverIO_temp[0] > 100) {
```
What is this caps vs no caps stuff here?

EUdds reviewed (changes requested):
Good work, small things left

EUdds commented on Frontend/src/Components/Faults/FaultsView.js:
```diff
@@ -1,13 +1,394 @@
+            eStopStrings.push("Battery");
+        }
+        if(_checkBooleanData("driver_eStop")) {
+            eStopStrings.push((eStopStrings.length == 0) ? "Driver" : "driver");
```
Same comment about this caps vs no caps stuff? What is that?

jrvollmer committed:
Image sizes adjust with height

jrvollmer committed:
Fixed some small bugs and implemented suggestions
